backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -221,48 +213,105 @@ def revoke_share(cls, engine: Engine, share_uri: str):
             )
             log.info(f'revoking s3 buckets succeeded = {revoked_s3_buckets_succeed}')
 
-            if source_environment.AwsAccountId != target_environment.AwsAccountId:
+            processor = DataSharingService.create_lf_processor(
+                session=session,
+                dataset=dataset,
+                share=share,
+                shared_tables=[],
+                revoked_tables=revoked_tables,
+                source_environment=source_environment,
+                target_environment=target_environment,
+                env_group=env_group,
+            )
+
+            if processor:
+                log.info(f'Revoking permissions to tables: {revoked_tables}')
+                revoked_tables_succeed = processor.process_revoked_shares()
+                log.info(f'revoking tables succeeded = {revoked_tables_succeed}')
+            else:
+                revoked_tables_succeed = False
+
+            existing_shared_items = ShareObjectRepository.check_existing_shared_items_of_type(
+                session,
+                share_uri,
+                ShareableType.Table.value
+            )
+            log.info(f'Still remaining LF resources shared = {existing_shared_items}')
+            if not existing_shared_items and revoked_tables:
+                log.info("Clean up LF remaining resources")
+                clean_up_tables = processor.delete_shared_database()
+                log.info(f"Clean up LF successful = {clean_up_tables}")
+
+            existing_pending_items = ShareObjectRepository.check_pending_share_items(session, share_uri)
+            if existing_pending_items:
+                new_share_state = share_sm.run_transition(ShareObjectActions.FinishPending.value)
+            else:
+                new_share_state = share_sm.run_transition(ShareObjectActions.Finish.value)
+            share_sm.update_state(session, share, new_share_state)
+
+            return revoked_folders_succeed and revoked_s3_buckets_succeed and revoked_tables_succeed
+
+    @staticmethod
+    def create_lf_processor(session,
+                            dataset,
+                            share,
+                            shared_tables,
+                            revoked_tables,
+                            source_environment,
+                            target_environment,
+                            env_group):
+        try:
```

Comment:
In general I think data_sharing_service should include minimal "sharing actions", instead the processors should be the ones executing the particular sharing mechanisms for approve and revoke. 
- `service` initializes the `processors` and calls `approve` or `revoke`
- `processor` defines `approve` or `revoke` workflow using `manager` functions
- `manager` defines actions that grant/revoke/configure the share. Typically use a AWS service
- `x_service_client` defines the boto3 calls as abstract as possible

Reply:
Following these principles, we should:
- Ultra minor: `def get_source_catalog(self):` rename to  `def get_catalog(self):` and remove references to source account because it depends on the GlueClient init values
- with the changes in the [PR](https://github.com/data-dot-all/dataall/pull/1016) for simplify db names we do not need to define the source account in the processor initialization. I suggest we do not pass the `catalog_details` from the sharing_service. Instead we should include this block of code directly in `lf_share_manager`

Reply:
Hi @dlpzx , I have refactored the code as per your recommendations and the code structure that is followed